internal/oci/runtime_vm.go

```diff
@@ -255,9 +255,15 @@ func (r *runtimeVM) StartContainer(c *Container) error {
 	go func() {
 		_, err := r.wait(r.ctx, c.ID(), "")
 		if err == nil {
-			if err1 := r.UpdateContainerStatus(c); err1 != nil {
+			if err1 := r.updateContainerStatus(c); err1 != nil {
 				logrus.Warningf("error updating container status %v", err1)
 			}
+
+			if c.state.Status == ContainerStateStopped {
```

Comment:
Do we have to clean up the container here or we can wait for kubelet to notice the status from ContainerStatus and then call StopContainer and RemoveContainer?

Reply:
That's a really good question.

At this point, StopContainer() has been called already, we sent the signal down to kata.
virtiofs & qemu processes are stopped, but containerd-shim-kata-v2 will still be around and will be consuming a high amount of CPU, which doesn't happen at all when using containerd.

That's the way I, with a not so good understanding of the whole architecture, could come up with what may be a reasonable solution.

I'm 100% open to a different approach if this one is wrong. :-)

Reply:
Thanks! I think it's fine to do the cleanup here to clean up heavy processes here and then leave the rest of the cleanup to the subsequent StopContainer call from the kubelet :)